nodeodm_client/nodeodm_client.py

```diff
@@ -0,0 +1,233 @@
+        with open(zip_path, 'wb') as f:
+            f.write(content)
+        self.logger.log_info(f"Results downloaded to {zip_path}")
+
+    async def process_task(self, task_id, transaction_image_dir):
+        """ Processes a task on NodeODM by uploading images, waiting for the task to finish and downloading the results.
+
+        :param task_id: The task ID for the NodeODM process
+        :param transaction_image_dir: The directory containing the images for the transaction
+        :return:
+        """
+        try:
+            uploaded_successfully = self.task_new_upload(task_id, transaction_image_dir)
+            if not uploaded_successfully:
+                self.logger.log_warning(f"Uploading images failed for task {task_id}.")
+                return None
+            self.task_new_commit(task_id)
+            status_code = 0
+            while status_code != 40 and status_code != 30 and status_code != 50:
+                status_code = self.get_task_status(task_id)
+                self.logger.log_info(switch_case(status_code))
+                if status_code is None:
+                    self.logger.log_warning("Error getting task status.")
+                    return None
+                await asyncio.sleep(60)
+            if status_code == 40:
+                self.download_results(task_id, transaction_image_dir)
+                self.remove_task(task_id)
+        except Exception as e:
+            self.logger.log_warning(f"An error occurred: {str(e)}")
+
+    async def calculate_orthophoto(self, transaction_id):
+        """
+        Calculates an orthophoto for a given transaction ID. This function creates a new task on NodeODM and uploads the images.
+        The function is not waiting for the task to finish, but returns the task ID.
+
+        :param transaction_id: The transaction ID of the image set to calculate the orthophoto for
+        :return: The task ID of the created task or None if an error occurred
+        """
+        try:
+            os.mkdir('logs')
+        except FileExistsError:
+            pass
+        try:
+            task_options = [
+                {"name": "fast-orthophoto", "value": "true"},
+                {"name": "skip-3dmodel", "value": "true"}
+            ]
+            transaction_image_dir = os.path.join(self.image_dir, transaction_id)
+            if not os.path.exists(transaction_image_dir):
+                self.logger.log_warning(f"Image directory {transaction_image_dir} does not exist. Exiting.")
+                return None
+            task_id = self.create_task_new_init(task_options)
+            if task_id is None:
+                self.logger.log_warning("Calculating the orthophoto failed.")
+                return None
+            asyncio.create_task(self.process_task(task_id, transaction_image_dir))
+            return task_id
+
+        except Exception as e:
+            self.logger.log_warning(f"An error occurred: {str(e)}")
```

Comment:
D.h. in dem Fall würde der aufrufende Prozess immer zurückbekommen, dass es keine Bilder für den S3 Bucket gibt?